server/crawler/crawler.go

```diff
@@ -0,0 +1,183 @@
+	var repoUpdated, repoRemoved int
+
+	allRepositories, err := getKurtosisPackageRepositories(ctx, githubClient)
+	if err != nil {
+		return 0, 0, stacktrace.Propagate(err, "Error search for Kurtosis package repositories on Github")
+	}
+
+	existingPackages := map[store.KurtosisPackageIdentifier]bool{}
+	for _, repository := range allRepositories {
+		kurtosisPackageContent, ok, err := extractKurtosisPackageContent(ctx, githubClient, repository)
+		if err != nil {
+			return 0, 0, stacktrace.Propagate(err, "An unexpected error occurred retrieving content for Kurtosis package repository '%s'",
+				repository.GetName())
+		}
+		if !ok {
+			logrus.Warnf("Kurtosis package repository '%s' content could not be retrieved. Does it contain a valid "+
+				"'%s' file at the root of the repository?", repository.GetFullName(), kurtosisYamlFileName)
+			continue
+		}
+
+		kurtosisPackageApi := api_constructors.NewKurtosisPackage(
+			string(kurtosisPackageContent.Identifier),
+		)
+		if err = crawler.store.UpsertPackage(kurtosisPackageApi); err != nil {
+			logrus.Errorf("An error occurred updating the package '%s' in the indexer store. Stored package data"+
+				"might be out of date until next refresh.", kurtosisPackageContent.Identifier)
+		}
+		existingPackages[kurtosisPackageContent.Identifier] = true
+		repoUpdated += 1
+	}
+
+	// remove all packages from the store which don't exist anymore
+	storedPackages, err := crawler.store.GetKurtosisPackages()
+	if err != nil {
+		return 0, 0, stacktrace.Propagate(err, "Unable to retrieve all Kurtosis packages currently stored by the indexer")
+	}
+	for _, storedPackage := range storedPackages {
+		packageIdentifier := store.KurtosisPackageIdentifier(storedPackage.GetName())
+		if _, found := existingPackages[packageIdentifier]; !found {
+			repoRemoved += 1
+			logrus.Infof("Removing package '%s' from the store as it doesn't seem to exist anymore", packageIdentifier)
+			if err = crawler.store.DeletePackage(packageIdentifier); err != nil {
+				logrus.Errorf("Unable to remove package '%s' from the store. This package data will be outdated", packageIdentifier)
+			}
+		}
+	}
+	return repoUpdated, repoRemoved, nil
+}
+
+func getKurtosisPackageRepositories(ctx context.Context, client *github.Client) ([]*github.Repository, error) {
+	// See https://docs.github.com/en/rest/search/search?apiVersion=2022-11-28#search-repositories
+	// Pagination logic taken from https://github.com/google/go-github/tree/master#pagination
+
+	var allRepositories []*github.Repository
+	repoSearchOpts := &github.SearchOptions{
+		Sort:      "stars",
+		Order:     "desc",
+		TextMatch: false,
+		ListOptions: github.ListOptions{
+			Page:    1,
+			PerPage: githubPageSize,
+		},
+	}
+	for {
+		repoSearchResult, rawResponse, err := client.Search.Repositories(ctx, "kurtosis-package in:topics template:false archived:false", repoSearchOpts)
```

Comment:
Does the package creator needs anything extra for search query to pick their package since search based of kurtosis.yml is not doable?

Reply:
They just need to add the `kurtosis-package` tag on their repo, and that should be enough